### ISSUE TITLE
Add API response code 201 to the valid list

### ISSUE THREAD

**Issue by mbruzek** [whitmo/charm-kubernetes#17]
The kubernetes charm failed a Juju deployment because the response from the server was 201.

Here is the log information:

```
2015-04-06 16:57:39 INFO unit.kubernetes/1.api-relation-changed logger.go:40 Registering machine
2015-04-06 16:57:39 INFO unit.kubernetes/1.api-relation-changed logger.go:40 {
2015-04-06 16:57:39 INFO unit.kubernetes/1.api-relation-changed logger.go:40   "kind": "Minion",
2015-04-06 16:57:39 INFO unit.kubernetes/1.api-relation-changed logger.go:40   "id": "10.0.0.173",
2015-04-06 16:57:39 INFO unit.kubernetes/1.api-relation-changed logger.go:40   "uid": "088a9631-dc7e-11e4-bb05-fa163ef99c53",
2015-04-06 16:57:39 INFO unit.kubernetes/1.api-relation-changed logger.go:40   "creationTimestamp": "2015-04-06T16:57:39Z",
2015-04-06 16:57:39 INFO unit.kubernetes/1.api-relation-changed logger.go:40   "selfLink": "/api/v1beta1/minions/10.0.0.173",
2015-04-06 16:57:39 INFO unit.kubernetes/1.api-relation-changed logger.go:40   "apiVersion": "v1beta1",
2015-04-06 16:57:39 INFO unit.kubernetes/1.api-relation-changed logger.go:40   "resources": {},
2015-04-06 16:57:39 INFO unit.kubernetes/1.api-relation-changed logger.go:40   "status": {
2015-04-06 16:57:39 INFO unit.kubernetes/1.api-relation-changed logger.go:40     "nodeInfo": {
2015-04-06 16:57:39 INFO unit.kubernetes/1.api-relation-changed logger.go:40       "machineID": "",
2015-04-06 16:57:39 INFO unit.kubernetes/1.api-relation-changed logger.go:40       "systemUUID": ""
2015-04-06 16:57:39 INFO unit.kubernetes/1.api-relation-changed logger.go:40     }
2015-04-06 16:57:39 INFO unit.kubernetes/1.api-relation-changed logger.go:40   }
2015-04-06 16:57:39 INFO unit.kubernetes/1.api-relation-changed logger.go:40 }
2015-04-06 16:57:39 INFO unit.kubernetes/1.api-relation-changed logger.go:40 Response status:201 reason:Created body:{u'status': {u'nodeInfo': {u'systemUUID': u'', u'machineID': u''}}, u'kind': u'Minion', u'uid': u'088a9631-dc7e-11e4-bb05-fa163ef99c53', u'apiVersion': u'v1beta1', u'resources': {}, u'creationTimestamp': u'2015-04-06T16:57:39Z', u'id': u'10.0.0.173', u'selfLink': u'/api/v1beta1/minions/10.0.0.173'}
2015-04-06 16:57:39 INFO unit.kubernetes/1.api-relation-changed logger.go:40 Registration error
...
2015-04-06 16:57:39 INFO unit.kubernetes/1.api-relation-changed logger.go:40 RuntimeError: Unable to register machine with {'Kind': 'Minion', 'metadata': {'name': u'10.0.0.173'}, 'ID': '10.0.0.173', 'resources': {'capacity': {'mem': '2049972 K', 'cpu': 2}}, 'HostIP': '10.0.0.173'}
2015-04-06 16:57:39 INFO juju.worker.uniter.context context.go:359 handling reboot
2015-04-06 16:57:39 ERROR juju.worker.uniter.operation runhook.go:86 hook "api-relation-changed" failed: exit status 1
```

According to the kubernetes API documentation: https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/api-conventions.md

201 appears to be a valid success code.

The kubernetes charm code only checks for 200, 202, and 409.  I think we should add 201.

**Comment by lazypower** [comment]
That seems correct. :+1:

**Comment by mbruzek** [comment]
Addressed by PR #18
